azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_action.go

```diff
@@ -0,0 +1,192 @@
+	if err != nil {
+		return fmt.Errorf("listing callback URL for Logic App Trigger %q (Resource Group %q / Workspace %q): %+v", triggerName, ruleID.ResourceGroup, ruleID.Workspace, err)
+	}
+
+	param := securityinsight.ActionRequest{
+		ActionRequestProperties: &securityinsight.ActionRequestProperties{
+			TriggerURI:         tresp.Value,
+			LogicAppResourceID: utils.String(lappId.ID(subscriptionId)),
+		},
+	}
+
+	if _, err := client.CreateOrUpdateAction(ctx, ruleID.ResourceGroup, OperationInsightsRPName, ruleID.Workspace, ruleID.Name, name, param); err != nil {
+		return fmt.Errorf("creating Sentinel Alert Rule Action %q (Resource Group %q / Workspace %q / Rule %q): %+v", name, ruleID.ResourceGroup, ruleID.Workspace, ruleID.Name, err)
+	}
+
+	resp, err = client.GetAction(ctx, ruleID.ResourceGroup, OperationInsightsRPName, ruleID.Workspace, ruleID.Name, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Sentinel Alert Rule Action %q (Resource Group %q / Workspace %q / Rule %q): %+v", name, ruleID.ResourceGroup, ruleID.Workspace, ruleID.Name, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Sentinel Alert Rule Action %q (Resource Group %q / Workspace %q / Rule %q)", name, ruleID.ResourceGroup, ruleID.Workspace, ruleID.Name)
+	}
+
+	id, err := parse.SentinelAlertRuleActionID(*resp.ID)
+	if err != nil {
+		return err
+	}
+	d.SetId(id.ID(subscriptionId))
+
+	return resourceArmSentinelAlertRuleActionRead(d, meta)
+}
+
+func resourceArmSentinelAlertRuleActionRead(d *schema.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Sentinel.AlertRulesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SentinelAlertRuleActionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetAction(ctx, id.ResourceGroup, OperationInsightsRPName, id.Workspace, id.Rule, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Sentinel Alert Rule Action %q was not found in Rule %q in Workspace %q in Resource Group %q - removing from state!", id.Name, id.Rule, id.Workspace, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Sentinel Alert Rule Action %q (Resource Group %q / Workspace %q / Rule %q): %+v", id.Name, id.ResourceGroup, id.Workspace, id.Rule, err)
+	}
+
+	d.Set("name", id.Name)
+
+	ruleID := parse.NewSentinelAlertRuleID(id.ResourceGroup, id.Workspace, id.Rule)
+	d.Set("sentinel_alert_rule_id", ruleID.ID(subscriptionId))
+
+	if prop := resp.ActionResponseProperties; prop != nil {
+		d.Set("logic_app_id", prop.LogicAppResourceID)
+		// TODO: Uncomment below line once https://github.com/Azure/azure-rest-api-specs/issues/9424 is addressed.
+		//d.Set("logic_app_trigger_name", prop.TriggerUrl)
```

Comment:
Until this bug is fixed i think this resource is blocked?

Reply:
Currently, users will have to use the lifecycle metadata to ignore the `logic_app_trigger_name`, also the import doesn't work. 
So we might had better to wait the API issue is fixed. However, on the other hand, the swagger issue has been hanging for a relative long time, and I don't think someone will pick it up soon.

Reply:
Do we really need the `azurerm_sentinel_alert_rule_action` resource to be so complex? All it really does is provide a link between a Logic app and a Sentinel alert rule, it should therefore be enough with two values.

This PR proposes:
```
resource "azurerm_sentinel_alert_rule_action" "example" {
  name                   = "example-alert-rule-action"
  sentinel_alert_rule_id = azurerm_sentinel_alert_rule_scheduled.example.id
  logic_app_id           = azurerm_logic_app_trigger_custom.example.logic_app_id
  logic_app_trigger_name = azurerm_logic_app_trigger_custom.example.name
  depends_on             = [azurerm_logic_app_trigger_custom.example]
}
```

I suggest we simplify it to:
```
resource "azurerm_sentinel_alert_rule_action" "example" {
  sentinel_alert_rule_id = azurerm_sentinel_alert_rule_scheduled.example.id

// One of the value below:
  logic_app_id           = azurerm_logic_app_trigger_custom.example.logic_app_id
  logic_app_trigger_name = azurerm_logic_app_trigger_custom.example.name
}
```

It also seems incorrect to set the `logic_app_trigger_name` to the value of `TriggerUrl`